projects/extension/ai/semantic_catalog.py

```diff
@@ -0,0 +1,433 @@
+    WHERE  attrelid = {plpy.quote_literal(relation)}::regclass
+    AND    attnum > 0
+    AND    NOT attisdropped;
+    """)
+    for r in result:
+        column_names.append(r["col"])
+    lines = get_obj_description(plpy, relation).split("\n")
+    filtered_lines = [
+        line
+        for line in lines
+        if not any(line.strip().lstrip("/* ").startswith(col) for col in column_names)
+    ]
+    obj_description = "\n".join(filtered_lines)
+
+    columns = []
+    system_prompt = """
+    You are a SQL expert generates natural language descriptions of columns of tables and views in a database.
+
+    The descriptions that you generate should be a concise single sentence.
+    You should include case sensitivity in your description if it is relevant.
+    """
+    message_content = f"""
+    Given the following table or view and some sample rows from it, generate a natural language description of all columns:
+
+    {obj_description}
+    {render_sample(plpy, relation)}
+    """
+    messages = [{"role": "user", "content": message_content}]
+    tools = [
+        {
+            "name": "generate_description",
+            "description": "Record description of the columns of the table or view in well-formed JSON",
+            "input_schema": {
+                "type": "object",
+                "properties": {
+                    "columns": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "name": {
+                                    "type": "string",
+                                    "description": "Name of the column",
+                                },
+                                "description": {
+                                    "type": "string",
+                                    "description": "Description of the column",
+                                },
+                            },
+                        },
+                    }
+                },
+            },
+        }
+    ]
+    provider = parsed_config.get("provider", None)
+    if provider == "anthropic":
+        model = parsed_config.get("model", "claude-3-5-sonnet-latest")
+        plan = plpy.prepare(
+            """
+                SELECT ai.anthropic_generate(
+                    $1
+                    , $2
+                    , system_prompt => $3
+                    , tools => $4
+                    , tool_choice => '{{"type": "tool", "name": "generate_description"}}'::jsonb
+                )
+            """,
+            ["text", "jsonb", "text", "jsonb"],
+        )
+        result = plpy.execute(
+            plan, [model, json.dumps(messages), system_prompt, json.dumps(tools)]
+        )
+        columns = json.loads(result[0]["anthropic_generate"])["content"][0]["input"][
+            "columns"
+        ]
+
+    elif provider == "ollama":
+        pass
```

Comment:
raise exception

Reply:
Done in 27853696d476c88c7c8ae47fa003081e711dc8ed